### PR TITLE
Problem 7: getDivisors

### PR DESCRIPTION
**Pull request by seemcat**
Input: a number
Return: largest divisor of number

Input: 10
Return: 5

### CODE REVIEW

File: solutions/7.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+const solution = (num, i = 2) => {
+  const checkPrime = (n) => {
+    for (i = 2; i <= n; i++) {
```
Comment (kinuesantos) [inline solutions/7.js]: use recursion instead of for loop

Comment (songz) [reply solutions/7.js]: WOOOOOW! NICE TRY @seemcat

File: solutions/7.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+const solution = (num, i = 1, bigD = -1) => {
+  if (i < num) {
+    if (num % i == 0) {
```
Comment (evo777) [inline solutions/7.js]: You can do `num % == 0 && bigD < i` and take out the line with `if (bigD < i)`. Just to reduce the number of line of code.

Comment (songz) [reply solutions/7.js]: why do you need to check if `bigD < 1 `?

Comment (evo777) [reply solutions/7.js]: I am not checking if `bigD` is less than 1. Aren't we checking to see if `bigD` is less than the current index, `i`? If it is, then we reassign the value of `bigD` to `i`, right?

File: solutions/7.js [hunk 3]
```diff
@@ -0,0 +1,10 @@
+const solution = (num, i = 1, bigD = -1) => {
+  if (i < num) {
+    if (num % i == 0 && bigD < i) {
```
Comment (evo777) [inline solutions/7.js]: No matter what, `i` will always be bigger than `bigD`. Therefore, you can just remove `bigD < i` logic

**Review by evo777 — APPROVED** [review]

**Pull request merged**